sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_link_async.py

```diff
@@ -244,6 +244,9 @@ async def _incoming_detach(self, frame) -> None:
             self._error = error_cls(condition=frame[2][0], description=frame[2][1], info=frame[2][2])
             await self._set_state(LinkState.ERROR)
         else:
+            if self.state != LinkState.DETACH_SENT:
```

Comment:
maybe we can add comments above this to describe what scenario we're accounting for/why we're doing this.

Reply:
yeah since this one is a "try again" to get the error, first one should be unknown as we dont know why it happened.